packages/wrangler/src/metrics/metrics-usage-headers.ts

```diff
@@ -0,0 +1,24 @@
+import { getMetricsConfig } from "./metrics-config";
+
+/**
+ * Add an additional header to publish requests if the user has opted into sending usage metrics.
+ *
+ * This allows us to estimate the number of instances of Wrangler that have opted-in
+ * without breaking our agreement not to send stuff if you have not opted-in.
+ */
+export async function getMetricsUsageHeaders(
```

Comment:
we're already logging an event on publish, so why this too?

Reply:
Perhaps the changeset for this was not clear enough... 
The idea is that by marking the publish/dev CF API calls with a header that says if metrics is enabled or not, we can estimate the proportion of wrangler users/projects that have opted-in to sending metrics, while never breaking our promise not to send things if they have opted-out.